### PR TITLE
Feature/proc data

### PR DESCRIPTION
**Pull request by Wikitelmex**
# requirements
- add model classes for all entities
- add associations between models
- add method in users that return 3 recent posts from user
- add a method that updates the post counter in the user when a post is saved
- add a method that returns the 5 most recent posts from a user
- add a method that updates the comments counter in the post when a comment is saved
- add a method that update the likes counter in the post when a like is saved

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @user_index = params[:user_id]
+  end
```
Comment (rloterh) [inline app/controllers/posts_controller.rb:5]: - [x] ApplicationController file `app/controllers/application_controller.rb` should contain `#current_user` method. Kindly improve upon this by adding code like below:
```ruby
def current_user
  User.first
end
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Alex, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦  🚦  

## Good Points 👍  
- Descriptive Pull Request
- Good ReadMe
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Wikitelmex) [reply app/controllers/posts_controller.rb:5]: thanks a lot, I just forgot that point

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    user = User.find(user_id)
+    user.increment!(:postscounter)
+    user.save
+  end
+
+  after_save :update_counter
+
+  def recent_comments
+    Comment.where(post_id: id).order(created_at: :desc).limit(5)
+  end
```
Comment (Dipeshtwis) [inline app/models/post.rb:16]: [OPTIONAL]

- So nice to see that you have used one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_counter` method every time a new post has been saved to the database. 

But now you can make the `update_counter` method private which will increase the security.

```ruby
# app/models/post.rb
  after_save :update_counter

  private

  def update_counter
    user.increment!(:posts_counter)
  end
```

**Review by Dipeshtwis — APPROVED** [review]
# Hi @Wikitelmex, 

### Status: Approved ✔️

:+1: You have done well implementing the changes. Well Done.

Your project is complete!  There are some optional changes I've suggested to you to make your project best, other than that nothing else to say.
Congratulations! 🎉 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
 **_Check the comments under the review_**


**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
-----

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Dipesh Kumar_**

------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**